packages/shared/src/components/fields/CodeField.tsx

```diff
@@ -0,0 +1,66 @@
+import type { ReactElement, ClipboardEventHandler, ChangeEvent } from 'react';
+import React, { useRef, useState } from 'react';
+import { TextField } from './TextField';
+
+interface CodeFieldProps {
+  onChange: (code: string) => void;
+  length?: number;
+}
+
+const DEFAULT_LENGTH = 6;
+
+export function CodeField({
+  onChange,
+  length = DEFAULT_LENGTH,
+}: CodeFieldProps): ReactElement {
+  const elementsRef = useRef<HTMLInputElement[]>(Array(length).fill(null));
+  const [code, setCode] = useState<string[]>(Array(length).fill(''));
+
+  const onPaste: ClipboardEventHandler<HTMLInputElement> = (e) => {
+    const text = e.clipboardData.getData('text');
+    const sliced = text.slice(0, DEFAULT_LENGTH);
+    setCode(sliced.split(''));
```

Comment:
We should validate that its only numbers in clipboard

Reply:
The `pattern` from the `input` element will make sure non-numeric is not accepted. So if you copy-pasted a string then definitely you copied the wrong code 🚀

Reply:
Yeah, but you can still paste anything, which is weird, usually this kind of inputs don't allow it

<img width="455" alt="image" src="https://github.com/user-attachments/assets/923c0116-cb52-4e87-af76-52881a147543" />

Reply:
I can also type non numbers for some reason as well

Reply:
Updated it to a `type="number"` instead.